src/simplejavacalculator/UI.java

```diff
@@ -33,11 +33,11 @@
     private final JTextArea text;
     private final JButton but[], butAdd, butMinus, butMultiply, butDivide,
             butEqual, butCancel, butSquareRoot, butSquare, butOneDevidedBy,
-            butCos, butSin, butTan, butxpowerofy, butlog, butrate;
+            butCos, butSin, butTan, butxpowerofy, butlog, butrate, butabs;
     private final Calculator calc;
 
     private final String[] buttonValue = { "0", "1", "2", "3", "4", "5", "6",
-            "7", "8", "9" };
+            "7", "8", "9", "10" };
```

Comment:
Why did you add "10", here? Anything related to `abs` feature?

Reply:
i'm terribly sorry about that. it was late at night, so my tired mind thought each button was assigned a value. however, looking back now, i see they don't. i assume we can discard that portion of the code.

Reply:
No worries @ItzSgtTanker! That happens sometimes 😃